File: python-neo4j-driver/common_friends.py
from neo4j import GraphDatabase, basic_auth
import neo4j_config

cfg = neo4j_config.neo4jConfig("../config.yml")
driver = GraphDatabase.driver("bolt://" + cfg['host'], auth=basic_auth(cfg['userName'], cfg['password']))
session = driver.session()

# Common friends

common_friends_query = """
MATCH (user:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf:Person)
WHERE user.name = {user} AND foaf.name = {foaf}
RETURN friend.name AS friend
"""

results = session.run(common_friends_query, parameters={"user": "Joe", "foaf": "Sally"})
for record in results:
    print(record["friend"])

session.close()

File: node-neo4j-driver/common_friends.js
const getNeo4jDriver = require("./yaml-config");
const driver = getNeo4jDriver("./config.yml");
const session = driver.session();

const commonFriendsQuery =
"MATCH (user:Person)-[:KNOWS]-(friend)-[:KNOWS]-(foaf:Person) \
 WHERE user.name = {name1} AND foaf.name = {name2} \
 RETURN friend.name AS friend";

session
    .run(commonFriendsQuery, {name1: "Joe", name2: "Sally"})
    .then(function (result) {
        session.close();
        result.records.forEach(function (record) {
            console.log(record.get('friend'));
        });
        driver.close();
    })
    .catch(function (error) { 
        console.log(error);
    });
